Source/WebSocket/hcwebsocket.cpp

```diff
@@ -69,12 +69,19 @@ HRESULT HC_WEBSOCKET::Connect(
             {
                 auto thisPtr{ static_cast<HC_WEBSOCKET*>(async->context) };
                 HRESULT hr = HCGetWebSocketConnectResult(async, &thisPtr->m_connectResult);
-                
-                if (SUCCEEDED(hr) && SUCCEEDED(thisPtr->m_connectResult.errorCode))
+
+                bool doDisconnect{ false };
+                bool doDecRef { false };
```

Comment:
Could you add comments to these to bools explaining the situations where we should/shouldn't disconnect and decRef? 

i.e. we auto disconnect if the client has already closed all handles to the socket and we release the provider's reference if we don't expect any more callbacks from the WebSocket provider